### ISSUE TITLE
`normalized_encode` incorrectly replaces `%3A` and `%2F` in path

### ISSUE THREAD

**Issue by davidtaylorhq** [sporkmonger/addressable#472]
Given a valid URL like `https://example.com/article/id%3A1.2%2F1/bar`, addressable's normalization replaces `%3A` with a literal `:`, and `%2F` with a literal `/`. Both of these replacements change the URL, meaning that requests for the URL may fail.

```ruby
> Addressable::URI.normalized_encode("https://example.com/article/id%3A1.2%2F1/bar")
=> "https://example.com/article/id:1.2/1/bar"
```

Using `URI#normalize` has the same issue with `%3A`, but seems to preserve the `%2F` correctly:
```ruby
> Addressable::URI.parse("https://example.com/article/id%3A1.2%2F1/bar").normalize.to_s
=> "https://example.com/article/id:1.2%2F1/bar"
```

My understanding of [RFC3986](https://datatracker.ietf.org/doc/html/rfc3986#section-6.2.2.2) is that reserved characters (including `:` and `/`) should not be decoded during normalization.

**Comment by sporkmonger** [comment]
I agree w/ your reading of the spec, only unreserved characters should be getting decoded.

**Comment by dentarg** [comment]
I think this issue is the same as https://github.com/sporkmonger/addressable/issues/366 but I'll keep this one open until it has been addressed.
